packages/docregistry/src/context.ts

```diff
@@ -557,9 +557,13 @@ export class Context<
    * @param newName - the new name for the document.
    */
   private async _rename(newName: string): Promise<void> {
-    const splitPath = this.path.split('/');
+    const splitPath = this.localPath.split('/');
     splitPath[splitPath.length - 1] = newName;
-    const newPath = splitPath.join('/');
+    let newPath = PathExt.join(...splitPath);
+    const drivName = this._manager.contents.driveName(this.path);
+    if (drivName) {
+      newPath = `${drivName}:${newPath}`;
+    }
```

Comment:
Looks like a minor typo:

```suggestion
    const driveName = this._manager.contents.driveName(this.path);
    if (driveName) {
      newPath = `${driveName}:${newPath}`;
    }
```